trackpy/feature.py

```diff
@@ -552,30 +563,38 @@ def locate(raw_image, diameter, minmass=100., maxsize=None, separation=None,
                 # To avoid degrading performance, assume gamut is zero to one.
                 # Have you ever encountered an image of unnormalized floats?
                 raw_image = 1 - raw_image
-        image = bandpass(raw_image, noise_size, smoothing_size, threshold)
+        image, black_level, noise = bandpass(raw_image, noise_size,
+                                             smoothing_size, threshold, True)
     else:
         image = raw_image.copy()
     # Coerce the image into integer type. Rescale to fill dynamic range.
     if np.issubdtype(raw_image.dtype, np.integer):
         dtype = raw_image.dtype
     else:
-        dtype = np.uint8
-    image = scale_to_gamut(image, dtype)
+        dtype = np.uint16
```

Comment:
From talking with @danielballan I think this was a pure speed / memory usage thing.